src/utils/skylink.ts

```diff
@@ -0,0 +1,175 @@
+import base64 from "base64-js";
+import base32Encode from "base32-encode";
+import parse from "url-parse";
+import { CustomClientOptions } from "../client";
+import { trimForwardSlash, trimSuffix, trimUriPrefix } from "./string";
+
+/**
+ * Base custom options for methods hitting the API.
+ *
+ * @property [endpointPath] - The relative URL path of the portal endpoint to contact.
+ */
+export type BaseCustomOptions = CustomClientOptions & {
+  endpointPath?: string;
+};
+
+/**
+ * Parse skylink options.
+ *
+ * @property [fromSubdomain] - Whether to parse the skylink as a base32 subdomain in a URL.
+ * @property [includePath] - Whether to include the path after the skylink, e.g. /<skylink>/foo/bar.
+ * @property [onlyPath] - Whether to parse out just the path, e.g. /foo/bar. Will still return null if the string does not contain a skylink.
+ */
+export type ParseSkylinkOptions = {
+  fromSubdomain?: boolean;
+  includePath?: boolean;
+  onlyPath?: boolean;
+};
+
+type ParseSkylinkBase32Options = {
+  onlyPath?: boolean;
+};
+
+export const uriHandshakePrefix = "hns:";
+export const uriHandshakeResolverPrefix = "hnsres:";
+export const uriSkynetPrefix = "sia:";
+
+/**
+ * Converts the given base64 skylink to base32.
+ *
+ * @param skylink - The base64 skylink.
+ * @returns - The converted base32 skylink.
+ */
+export function convertSkylinkToBase32(skylink: string): string {
+  const decoded = base64.toByteArray(skylink.padEnd(skylink.length + 4 - (skylink.length % 4), "="));
+  return base32Encode(decoded, "RFC4648-HEX", { padding: false }).toLowerCase();
+}
+
+/**
+ * Returns the default base custom options for the given endpoint path.
+ *
+ * @param endpointPath - The endpoint path.
+ * @returns - The base custom options.
+ */
+export function defaultOptions(endpointPath: string): CustomClientOptions & { endpointPath: string } {
+  return {
+    endpointPath,
+    APIKey: "",
+    customUserAgent: "",
+  };
+}
+
+// TODO: This will be smarter. See
+// https://github.com/NebulousLabs/skynet-docs/issues/21.
+/**
+ * Returns the default portal URL.
+ *
+ * @returns - The portal URL.
+ */
+export function defaultPortalUrl(): string {
```

Comment:
I would put this in the url file.

Reply:
Good call. I wasn't sure where that one should go. PR updated.